libsigner/src/v0/messages.rs

```diff
@@ -542,47 +542,121 @@ impl StacksMessageCodec for MockBlock {
-    burn_block_height: u64,
-    current_miner_pkh: Hash160,
-    parent_tenure_id: ConsensusHash,
-    parent_tenure_last_block: StacksBlockId,
-    parent_tenure_last_block_height: u64,
-    active_signer_protocol_version: u64,
-    local_supported_signer_protocol_version: u64,
+    /// The tip burn block (i.e., the latest bitcoin block) seen by this signer
+    pub burn_block: ConsensusHash,
+    /// The tip burn block height (i.e., the latest bitcoin block) seen by this signer
+    pub burn_block_height: u64,
+    /// The signer's view of who the current miner should be (and their tenure building info)
+    pub current_miner: StateMachineUpdateMinerState,
+    /// The active signing protocol version
+    pub active_signer_protocol_version: u64,
+    /// The highest supported signing protocol by the local signer
+    pub local_supported_signer_protocol_version: u64,
+}
+
+/// Message for update the Signer State infos
+#[derive(Debug, Clone, PartialEq, Deserialize, Serialize)]
+pub enum StateMachineUpdateMinerState {
+    /// There is an active miner
+    ActiveMiner {
+        /// The pubkeyhash of the current miner's signing key
+        current_miner_pkh: Hash160,
+        /// The tenure ID of the current miner's active tenure
+        tenure_id: ConsensusHash,
+        /// The tenure that the current miner is building on top of
+        parent_tenure_id: ConsensusHash,
+        /// The last block of the parent tenure (which should be
+        ///  the block that the next tenure starts from)
+        parent_tenure_last_block: StacksBlockId,
+        /// The height of the last block of the parent tenure (which should be
+        ///  the block that the next tenure starts from)
+        parent_tenure_last_block_height: u64,
+    },
+    /// The signer doesn't believe there's any valid miner
+    NoValidMiner,
+}
+
+impl StateMachineUpdateMinerState {
+    fn get_variant_id(&self) -> u8 {
+        match self {
+            StateMachineUpdateMinerState::NoValidMiner => 0,
+            StateMachineUpdateMinerState::ActiveMiner { .. } => 1,
+        }
+    }
+}
+
+impl StacksMessageCodec for StateMachineUpdateMinerState {
+    fn consensus_serialize<W: Write>(&self, fd: &mut W) -> Result<(), CodecError> {
+        self.get_variant_id().consensus_serialize(fd)?;
+        match self {
+            StateMachineUpdateMinerState::ActiveMiner {
+                current_miner_pkh,
+                tenure_id,
+                parent_tenure_id,
+                parent_tenure_last_block,
+                parent_tenure_last_block_height,
+            } => {
+                current_miner_pkh.consensus_serialize(fd)?;
+                tenure_id.consensus_serialize(fd)?;
+                parent_tenure_id.consensus_serialize(fd)?;
+                parent_tenure_last_block.consensus_serialize(fd)?;
+                parent_tenure_last_block_height.consensus_serialize(fd)?;
+            }
+            StateMachineUpdateMinerState::NoValidMiner => return Ok(()),
```

Comment:
Nit: this may just be a personal preference, but it seems cleaner if this case is ` {}` and both reach the `Ok(())`.